src/tribler/core/utilities/tribler_process_manager.py

```diff
@@ -0,0 +1,175 @@
+from __future__ import annotations
+
+import logging
+import sqlite3
+import sys
+from functools import wraps
+from pathlib import Path
+from threading import Lock
+from typing import ContextManager, List, Optional
+
+from decorator import contextmanager
+
+from tribler.core.utilities.tribler_process import ProcessKind, TriblerProcess
+
+logger = logging.getLogger(__name__)
+
+DB_FILENAME = 'processes.sqlite'
+
+CREATE_SQL = """
+    CREATE TABLE IF NOT EXISTS processes (
+        rowid INTEGER PRIMARY KEY AUTOINCREMENT,
+        row_version INTEGER NOT NULL DEFAULT 0,
+        pid INTEGER NOT NULL,
+        kind TEXT NOT NULL,
+        "primary" INT NOT NULL,
+        canceled INT NOT NULL,
+        app_version TEXT NOT NULL,
+        started_at INT NOT NULL,
+        creator_pid INT,
+        api_port INT,
+        shutdown_request_pid INT,
+        shutdown_requested_at INT, 
+        finished_at INT,
+        exit_code INT,
+        error_msg TEXT,
+        error_info JSON,
+        other_params JSON
+    )
+"""
+
+
+global_process_manager: Optional[ProcessManager] = None
+
+_lock = Lock()
+
+
+def set_global_process_manager(process_manager: Optional[ProcessManager]):
+    global global_process_manager  # pylint: disable=global-statement
+    with _lock:
+        global_process_manager = process_manager
+
+
+def get_global_process_manager() -> Optional[ProcessManager]:
+    with _lock:
+        return global_process_manager
+
+
+def set_error(error_msg: Optional[str] = None, error_info: Optional[dict] = None,
+              exc: Optional[Exception] = None, replace: bool = False):
+    process_manager = get_global_process_manager()
+    if process_manager is None:
+        logger.warning('Cannot set error for process locker: no process locker global instance is set')
+    else:
+        process_manager.current_process.set_error(error_msg, error_info, exc, replace)
+        process_manager.save(process_manager.current_process)
+
+
+def with_retry(func):
+    @wraps(func)
+    def new_func(*args, **kwargs):
+        try:
+            return func(*args, **kwargs)
+        except sqlite3.Error as e:
+            logger.warning(f'Retrying after the error: {e.__class__.__name__}: {e}')
+            return func(*args, **kwargs)
+    new_func: func
+    return new_func
+
+
+class ProcessManager:
+    def __init__(self, root_dir: Path, process_kind: ProcessKind, creator_pid: Optional[int] = None, **other_params):
+        self.root_dir = root_dir
+        self.filename = self._get_file_name(root_dir)
+        self.current_process = TriblerProcess.current_process(process_kind, creator_pid, **other_params)
+        self.primary_process = self.atomic_get_primary_process(process_kind, self.current_process)
+
+    @classmethod
+    def _get_file_name(cls, root_dir: Path) -> Path:  # The method is added for easier testing
```

Comment:
NIT:

This method could be replaced by the following construction (that should be same convenient for testing):

```python
class ProcessManager:
    def __init__(self, root_dir: Path, process_kind: ProcessKind, creator_pid: Optional[int] = None,
                 db_file_name: str = DB_FILENAME, **other_params):
        self.root_dir = root_dir
        self.filename = root_dir / DB_FILENAME
        self.current_process = TriblerProcess.current_process(process_kind, creator_pid, **other_params)
        self.primary_process = self.atomic_get_primary_process(process_kind, self.current_process)
```

Reply:
I removed the method